### PR TITLE
IDX Cluster Fixing and Feature extension

### PR DESCRIPTION
**Pull request by schneewe**
Due to current circumstances I wanted to use the IDX Cluster configuration playbook and therefor I have some changes.
Adding multisite option (new variables, extended tasks)
Fixed the searchhead configuration task and added to example playbook
Fixed Version check for splunk to use the right commands

### CODE REVIEW

**Comment by jewnix** [comment]
@schneewe Seems like you didn't merge the latest changes from the upstream before you created the PR.

File: roles/splunk/defaults/main.yml [hunk 1]
```diff
@@ -42,21 +51,28 @@ splunk_force_kill: False
...
 splunk_disable_mgmt_port: false # If set to true, will disable splunkd management port during installation
-splunk_mgmt_uri: "{{ ansible_fqdn }}" # If the `ansible_fqdn` is not resolvable by other hosts, you can set it to something like `ansible_facts.default_ipv4.address` in `host_vars` or `group_vars` to use the IP address instead.
+splunk_disable_kvstore: false # If set to true, will disable the kvstore
 splunkd_port: 8089 # If changed, will overwrite the default port number used by splunkd
 git_local_clone_path: ~/ # Base directory under which repositories for app deplyoment should be cloned to
 git_server: undefined # e.g. ssh://git@mygithost:1234 - Note that this may be set in an all.yml group_var or inside the git_apps dictionary within host_vars
-git_key: undefined # Path to SSH key for cloning repositories - Note that this may be set in an all.yml group_var or inside the git_apps dictionary within host_vars
+git_key: ~ # Path to SSH key for cloning repositories - Note that this may be set in an all.yml group_var or inside the git_apps dictionary within host_vars
```
Comment (jewnix) [inline roles/splunk/defaults/main.yml:58]: This is used [here](https://github.com/splunk/ansible-role-for-splunk/blob/master/roles/splunk/tasks/configure_apps.yml#L94)

File: roles/splunk/defaults/main.yml [hunk 2]
```diff
@@ -42,21 +51,28 @@ splunk_force_kill: False
...
 git_project: undefined
 git_version: master # Configure default version to clone, overridable inside the git_apps dictionary within host_vars
+splunk_app_deploy_path: etc/apps # Path under $SPLUNK_HOME/ to deploy apps to - Note that this may be set in group_vars, host_vars, playbook vars, or inside the git_apps dictionary within host_vars
```
Comment (jewnix) [inline roles/splunk/defaults/main.yml:61]: I'm not sure if changing it from `undefined` breaks anything, but it's being set [here](https://github.com/splunk/ansible-role-for-splunk/blob/master/roles/splunk/tasks/configure_apps.yml#L20) and in the `group_vars`

File: roles/splunk/defaults/main.yml [hunk 3]
```diff
@@ -13,11 +13,19 @@ splunk_package_url_uf: "https://download.splunk.com/products/universalforwarder/
...
-splunk_nix_user: splunk
-splunk_nix_group: splunk
 local_os_user: false # Whenther or not to force creation of a user using the `luseradd` or not.
 local_os_group: false # Whether or not to force creation of a group using the `lgroupadd` or not.
+splunk_nix_user: splunk
+splunk_nix_group: splunk
+splunk_server_sslVersions: tls1.2
+splunk_web_sslVersions: tls1.2
+splunk_webserver_port: 8000
+splunk_licenses: []
```
Comment (jewnix) [inline roles/splunk/defaults/main.yml:26]: there is a different variable for this [here](https://github.com/splunk/ansible-role-for-splunk/blob/master/roles/splunk/defaults/main.yml#L22)

File: roles/splunk/defaults/main.yml [hunk 4]
```diff
@@ -13,11 +13,19 @@ splunk_package_url_uf: "https://download.splunk.com/products/universalforwarder/
 splunk_download_local: true  # This defines how the download process works. If `true` it will download to localhost and copy around to hosts from there. If `false` each host will download the package individually.
 splunk_install_type: undefined # There are two ways to configure this. The easiest way is to nest hosts under either a "full" group or a "uf" group in your inventory and main.yml will handle it for you. Or, you can also set the value via a group_vars or host_vars file.
 splunk_install_path: /opt # Base directory on the operating system to which splunk should be installed
+splunk_file_type: tgz # Default is a tgz installer
```
Comment (jewnix) [inline roles/splunk/defaults/main.yml:16]: This is not used anywhere

File: roles/splunk/tasks/configure_idxc_manager.yml [hunk 5]
```diff
@@ -8,7 +8,18 @@
 # https://docs.splunk.com/Documentation/Splunk/latest/Indexer/Configuremanagerwithserverconf
 - name: Setting clustering mode based on Splunk version number
   set_fact:
-    mode_value: "{% if splunk_version_release | float < 8.1 %}master{% else %}manager{% endif %}"
+    mode_value: "{{ splunk_version_release is version('8.1', '<') | ternary('master', 'manager') }}"
+
+- name: Setting site affinity of cluster manager if not set
+  set_fact:
+    site_affinity: "{% if splunk_idxc_multisite and splunk_idxc_site_affinity == 'site0' %}site1{% else %}{{ splunk_idxc_site_affinity }}{% endif %}"
```
Comment (jewnix) [inline roles/splunk/tasks/configure_idxc_manager.yml:15]: `site0` is only available on search-heads or forwarders with indexer discovery

File: playbooks/splunk_idxc_deploy.yml [hunk 6]
```diff
@@ -21,3 +21,10 @@
     - ../roles/splunk
   vars:
     - deployment_task: configure_idxc_member.yml
+
+- hosts:
+    - search
```
Comment (jewnix) [inline playbooks/splunk_idxc_deploy.yml:26]: You should add `search` [here](https://github.com/splunk/ansible-role-for-splunk/blob/master/playbooks/splunk_idxc_deploy.yml#L5)

File: roles/splunk/defaults/main.yml [hunk 7]
```diff
@@ -42,21 +51,28 @@ splunk_force_kill: False
 systemd_unit_full: Splunkd # You can change this in `host_vars` or `group_vars` to customize the service name.
 systemd_unit_uf: SplunkForwarder # You can change this in `host_vars` or `group_vars` to customize the service name.
 splunk_disable_mgmt_port: false # If set to true, will disable splunkd management port during installation
-splunk_mgmt_uri: "{{ ansible_fqdn }}" # If the `ansible_fqdn` is not resolvable by other hosts, you can set it to something like `ansible_facts.default_ipv4.address` in `host_vars` or `group_vars` to use the IP address instead.
+splunk_disable_kvstore: false # If set to true, will disable the kvstore
```
Comment (jewnix) [inline roles/splunk/defaults/main.yml:54]: I'm not sure where this is being used

**Review by jewnix — COMMENTED** [review]
This PR needs a lot of work.

**Comment by schneewe** [comment]
Yeah I see, you are right, there was a different base used, I will create a complete new one.

**Comment by schneewe** [comment]
I created a new PR #195

**Pull request closed without merging**